src/data/bucket/line_bucket.js

```diff
@@ -231,15 +216,21 @@ class LineBucket implements Bucket {
     }
 
     addLine(vertices: Array<Point>, feature: BucketFeature, join: string, cap: string, miterLimit: number, roundLimit: number, index: number, imagePositions: {[string]: ImagePosition}) {
-        let lineDistances = null;
+        this.distance = 0;
+        this.scaledDistance = 0;
+        this.totalDistance = 0;
+
```

Comment:
Do you think creating a line for each object is a significant cost? putting per-line values directly on the bucket seems a bit less nice but probably ok if there is a significant benefit.

Reply:
@ansis we had `distance` as a bucket state before this PR, so the latter doesn't change the status quo in that sense — I moved the line stats to shared state too for simplicity/consistency, but I think we can revisit this in a follow-up PR.